data-generator/src/main/java/com/github/streamshub/kafka/data/generator/Main.java

```diff
@@ -0,0 +1,37 @@
+package com.github.streamshub.kafka.data.generator;
+
+import com.github.streamshub.kafka.data.generator.examples.ClickStreamData;
+import com.github.streamshub.kafka.data.generator.examples.SalesData;
+
+import java.util.Arrays;
+
+public class Main {
+    public static void main(String[] args) {
+        String bootstrapServers = System.getenv("KAFKA_BOOTSTRAP_SERVERS");
+        String[] dataTypes = System.getenv("DATA").split(",");
+        Arrays.stream(dataTypes)
+                .map(Main::getDataClass)
+                .forEach(data -> {
+                    System.out.println("Starting Data Generator " + data.getClass().getName() + "...");
+                    Thread dgThread = new Thread(new DataGenerator(bootstrapServers, data));
+                    dgThread.start();
+
+                    //So two classes don't fight when registering the schemas in the registry
```

Comment:
ah, this is what I hit on. Maybe you could arrange things so there is no async work. Have a Driver that knows about a list of DataGenerators. The Driver could be responsible for the looping and sleeping, every iteration it calls `generator.generate()` on all generators.

Reply:
(or give Main this responsibility)

Reply:
What I observed was it crashing due to the two threads interfering with each other while trying to register the schema, which I think it should only do at the beginning. Yeah perhaps we can start one data generator and send the first message, then return and let the second one start.

Reply:
I've re-read your comment and realised what you were suggesting, so have a single Kafka producer sends all the data, rather than running multiple Kafka producers in parallel right? I think that's a good idea

Reply:
Yes. One producer, or multiple isn't so important. As long as we remove the parallelism. Something like:

1. send message x to topic A
2. block until message x is acked
3. send message y to topic B
4. block until message y is acked

Not very efficient but with a couple of messages per 3 seconds it's not a problem yet.